### DISCUSSION TITLE
query on concurrency in benchmarking results

### DISCUSSION THREAD

**Discussion by pi194046** [TechEmpower/FrameworkBenchmarks#6742] [Q&A]
In the benchmarking results when concurrency is specified for 16,32,64,128,256,512 . Does this imply the server side concurrancy or client side conccurance in number of simultaneous connections . 

In case it implies server side consurrance , what is client ie wrk or load testing specificiations ie number of threads , number of simultaneous connections used by load testing client

In case it implies client side concurrency , what is concurrency of in the form of threads/processes on the client . 
In various frameworks mentioned , especially in java frameworks concurrancy can be achieved by multi threading or multi processing . In the benchmarks what is modality used by java application .

**Comment by billywhizz** [comment]
this is the number of client connections. the "-c" flag for wrk. so if concurrency is 16 and the machine wrk is running on has 16 threads (i think wrk will use a thread for each available core/hyperthread on host machine which on Citrine is 28 threads) then each thread will have 1 http client connection. if concurrency is 512 then each wrk thread will have 512 / 16 = 32 client connections.
